main.rkt

```diff
@@ -0,0 +1,218 @@
+; the event type, the variable will not exist.
+(define github-ref
+  (make-parameter (getenv "GITHUB_REF") #f 'github-ref))
+
+; Only set for pull request events. The name of the head branch.
+(define github-head-ref
+  (make-parameter (getenv "GITHUB_HEAD_REF") #f 'github-head-ref))
+
+; Only set for pull request events. The name of the base branch.
+(define github-base-ref
+  (make-parameter (getenv "GITHUB_BASE_REF") #f 'github-base-ref))
+
+; Returns the API URL. For example: `https://api.github.com`.
+(define github-api-url
+  (make-parameter (getenv "GITHUB_API_URL") #f 'github-api-url))
+
+; This doesn't support anything but getting stdout -- but that's OK for now!
+(define/guard (run-cmd cmd-name . args)
+  (define cmd-path (or (find-executable-path cmd-name)
+                       ; Racket doesn't know about $PATHEXT:
+                       (find-executable-path (string-append ".exe"))))
+  (guard (not cmd-path) then
+         (error (format "couldn't find ~a executable in PATH" cmd-name)))
+  (define-values (proc stdout stdin stderr)
+    (apply subprocess `(#f #f #f ,cmd-path ,@args)))
+  (close-output-port stdin)
+  (subprocess-wait proc)
+  (define exit-code (subprocess-status proc))
+  (define stdout-string (port->string stdout))
+  (define stderr-string (string-trim (port->string stderr)))
+  (close-input-port stdout)
+  (close-input-port stderr)
+  (guard (zero? exit-code) else
+         (error (format "command '~a ~a' exited with code ~a"
+                        cmd-path
+                        (string-join args " ")
+                        exit-code)))
+  (guard (zero? (string-length stderr-string)) else
+         (error (format "command '~a ~a' wrote to stderr: ~a"
+                        cmd-path
+                        (string-join args " ")
+                        stderr-string)))
+  stdout-string)
+
+(define (git-diff-names commitish)
+  (string-split (run-cmd "git" "diff" "--name-only" "-z" commitish)
+                "\0"))
+
+(define (git-path path)
+  (string-split (run-cmd "git" "ls-tree" "-r" "-z" "--name-only" "HEAD" path)
+                "\0"))
+
+(define (git-ref->pr-number ref)
+  (match ref
+    [(regexp #rx"^refs/pull/([0-9]+)/merge$" (list _ num))
+     (string->number num)]
+    [_
+     (error (format "ref ~a doesn't represent a pull request"))]))
+
+(define-record-type github-review-request
+  (owner-repo pull-number commit-id body event comments))
+
+(define (github-review-request-body-jsexpr req)
+  (match req
+    [(github-review-request #:commit-id commit-id
+                            #:body body
+                            #:event event
+                            #:comments comments)
+     (make-immutable-hash `(#;(commit_id . ,commit-id)
+                            (body . ,body)
+                            (event . ,event)
+                            (comments . ,(map github-review-comment-jsexpr comments))))]))
```

Comment:
Instead of using `make-immutable-hash` with quasiquoting, it's probably simpler to just use `hash`:

```scheme
(hash 'commit_id commit-id
      'body body
      'event event
      'comments (map ...))
```

(Same comment goes for the other use of `make-immutable-hash`.)